pkg/upgrademonitor/upgrade_monitor.go

```diff
@@ -335,20 +384,72 @@ func isSingletonCondition(singletonConditionTypes []mcfgv1.StateProgress, condit
 	return false
 }
 
+// UpdateMachineConfigNodeSpecDesiredAnnotations sets the desired config version and image
+// annotation values in the `Spec` of an existing MachineConfigNode resource
+func UpdateMachineConfigNodeSpecDesiredAnnotations(fgHandler ctrlcommon.FeatureGatesHandler, mcfgClient mcfgclientset.Interface, nodeName, desiredConfig, desiredImage string) error {
+	if fgHandler == nil {
+		return nil
+	}
+
+	// Check that the MachineConfigNode feature gate is enabled
+	if !fgHandler.Enabled(features.FeatureGateMachineConfigNodes) {
+		klog.Infof("MachineConfigNode FeatureGate is not enabled.")
+		return nil
+	}
+
+	// Get the existing MCN
+	mcn, mcnErr := mcfgClient.MachineconfigurationV1().MachineConfigNodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	// Note that this function is only intended to update the Spec of an existing MCN. We should
+	// not reach this point if there is not an existing MCN for a node, but we need to handle the
+	// DNE error situation just in case.
+	if mcnErr != nil {
+		// no existing MCN found since no resource found
+		if apierrors.IsNotFound(mcnErr) {
+			return fmt.Errorf("an MCN for %s node does not exits; skipping MCN desired annotations spec update", nodeName)
```

Comment:
nit: `exits` -> `exist`

Also since we'd be returning the error here anyways (and like you said, we shouldn't really ever get a DNE), maybe we don't need to special case this error and just have the return below?